module/enrichers.mjs

```diff
@@ -823,6 +903,36 @@ async function rollAction(event) {
         case "tool":
           options.ability = ability;
           return await actor.rollToolCheck(tool, options);
+        case "item":
+          if ( target.dataset.rollItemUuid ) {
+            const gameActor = await fromUuid(target.dataset.rollItemActor);
+            if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+              return (await fromUuid(target.dataset.rollItemUuid)).use();
+            } else {
+              return ui.notifications.warn(`You do not have ownership of ${gameActor?.name}, and cannot use this item.`);
+            }
+          } else if ( target.dataset.rollItemName ) {
+            if ( (canvas.tokens.controlled.length === 0 && !target.dataset.rollItemActor) || canvas.tokens.controlled[0].actor.items.getName(target.dataset.rollItemName) ) {
+              return dnd5e.documents.macro.rollItem(target.dataset.rollItemName);
+            } else if ( target.dataset.rollItemActor ) {
+              const gameActor = await fromUuid(target.dataset.rollItemActor);
+              if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+                if ( gameActor.items.getName(target.dataset.rollItemName) ) {
+                  if ( canvas.tokens.controlled.length > 0 ) {
+                    ui.notifications.warn(`Your controlled token ${canvas.tokens.controlled[0].name} does not have an Item with name ${target.dataset.rollItemName} using Item Enricher owner`);
+                  }
+                  return (await fromUuid(target.dataset.rollItemActor)).items.getName(target.dataset.rollItemName).use();
+                } else {
+                  return ui.notifications.warn(`${gameActor.name} does not have an Item with name ${target.dataset.rollItemName}.`);
+                }
+              } else {
+                return ui.notifications.warn(`You do not have ownership of ${gameActor?.name}, and cannot use this item.`);
+              }
+            } else {
+              return ui.notifications.warn(`Your controlled actor ${canvas.tokens.controlled[0]?.actor.name} does not have an Item with name ${target.dataset.rollItemName}.`);
+            }
+          }
+
```

Comment:
I find this logic a bit hard to follow, I've taken a pass in what I hope maintains the functionality. I've skipped going through the `rollItem` macro because I don't think it adds much here.

```suggestion
            actor = canvas.tokens.controlled[0]?.actor ?? game.user.character
              ?? await fromUuid(target.dataset.rollItemActor);
            if ( !actor ) {
              ui.notifications.error("No token or selected actor for which to use {item}.", { localize: true });
              return;
            }

            const item = actor?.items.getName(target.dataset.rollItemName);
            if ( !item ) {
              ui.notifications.error("{actor} does not have an Item with name {item}.", { localize: true });
              return;
            }

            return item.use();
```

Basically it sets the actor to be the first of (1) first controlled token on scene, (2) assigned actor, or (3) actor within which the enricher is embedded.

The notification messages will need to be converted to properly localized strings.

Reply:
> I find this logic a bit hard to follow  

Same, tbh.
This is mostly for handling enrichers in chat cards, and while I think it would be odd for Player1 to use an `[[/item Shortsword]]` enricher from an NPC's Multiattack chat card - to use their own Shortsword, not the NPC's - but I dont see why it should necessarily be prevented?  

With the proposed change, if a user does not own the actor the enricher came from, and does not have a token selected,  but the actor assigned to them does have an item of a matching name, the item from the actor that owns the enricher will be used, I dont think the owned actor will ever get used, so I would swap around line 916 to 
`?? game.user.character ?? await fromUuid(target.dataset.rollItemActor);`  

Then, if the selected token does not have an item of that name, it will just throw an error, and not fall back to the owner actor, which I think is fine but was a part of what I think was discussed earlier, and is the use of the warning in line 922